webapp-django/crashstats/crashstats/utils.py

```diff
@@ -235,10 +238,44 @@ def get_versions_for_product(product):
     for item in ret['facets']['version']:
         version = item['term']
         try:
+            # This generates the sort key but also parses the version to
+            # make sure it's a valid looking version
             versions.append((generate_version_key(version), version))
         except VersionParseError:
             pass
+
     versions.sort(key=lambda v: v[0], reverse=True)
+    versions = [v[1] for v in versions]
+
+    if use_cache:
+        # Cache value for an hour plus a fudge factor in seconds
+        cache.set(key, versions, (60 * 60) + random.randint(60, 120))
+
+    return versions
+
+
+def get_version_context_for_product(product):
+    """Returns version context for a specified product
+
+    This gets the versions for a product and generates a context consisting
+    of betas, featured versions, and versions.
+
+    If SuperSearch returns an error, this returns an empty list.
+
+    NOTE(willkg): This data can be noisy in cases where crash reports return
+    junk versions. We might want to add a "minimum to matter" number.
+
+    :arg product: the product to query for
+
+    :returns: list of version dicts sorted in reverse order or ``[]``
+
+    """
+    key = 'get_version_context_for_product:%s' % product.lower().replace(' ', '')
+    ret = cache.get(key)
+    if ret is not None:
+        return ret
+
+    versions = get_versions_for_product(product, use_cache=False)
```

Comment:
This passes `use_cache=False` so that when this goes to recompute things, it's not inadvertently using a cached version of the versions which would result in double-caching and a longer period of time for the return for this function to expire out of cache.